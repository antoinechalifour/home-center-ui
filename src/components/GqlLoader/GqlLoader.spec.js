import React from 'react'
import { shallow } from 'enzyme'
import gqlLoaderHoc from '.'

const Component = ({ data }) => <div>Hello {data.name}</div>

describe('components/GqlLoader', () => {
  it('Should render null when data is loader', () => {
    const GqlLoader = gqlLoaderHoc(Component)
    const props = {
      data: {
        loading: true,
        name: 'John'
      }
    }

    const wrapper = shallow(<GqlLoader {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Should render the component otherwise', () => {
    const GqlLoader = gqlLoaderHoc(Component)
    const props = {
      data: {
        loading: false,
        name: 'John'
      }
    }

    const wrapper = shallow(<GqlLoader {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
