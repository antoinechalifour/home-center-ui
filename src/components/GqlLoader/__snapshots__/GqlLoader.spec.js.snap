// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/GqlLoader Should render null when data is loader 1`] = `<Loader />`;

exports[`components/GqlLoader Should render the component otherwise 1`] = `
<Component
  data={
    Object {
      "loading": false,
      "name": "John",
    }
  }
/>
`;
